src/main/java/com/github/osisoft/ocs_sample_library_preview/sds/StreamsClient.java

```diff
@@ -1104,6 +1105,61 @@ public String getRangeValuesInterpolated(String tenantId, String namespaceId, St
         return response;
     }
 
+    /**
+     * gets values at the indexes provided
+     * 
+     * @param tenantId    tenant to work against
+     * @param namespaceId namespace to work against
+     * @param streamId    the stream to get values from
+     * @param index       the list of indexes to retrieve values for
+     * @return string of the array of values
+     * @throws SdsError any error that occurs
+     */
+    public String getIndexCollectionValues(String tenantId, String namespaceId, String streamId,
+            ArrayList<String> index) throws SdsError {
+        URL url = null;
+        HttpURLConnection urlConnection = null;
+        String response = "";
+
+        try {
+            String indexString = "";
+            for (int i = 0; i < index.size(); i++) {
+                if (i == 0)
+                    indexString = index.get(i);
+                else
+                    indexString += "&index=" + index.get(i);
+            }
+
+            String intermediate = getIndexCollectionQuery.replace("{apiVersion}", apiVersion)
+                    .replace("{tenantId}", tenantId).replace("{namespaceId}", namespaceId)
+                    .replace("{streamId}", streamId).replace("{index}", indexString);
+            url = new URL(baseUrl + intermediate);
+            urlConnection = baseClient.getConnection(url, "GET");
+
+            int httpResult = urlConnection.getResponseCode();
+
+            if (httpResult == HttpURLConnection.HTTP_OK || httpResult == HttpURLConnection.HTTP_CREATED) {
+            } else {
+                throw new SdsError(urlConnection, "get range of interpolated values request failed");
+            }
+
+            response = baseClient.getResponse(urlConnection);
+        } catch (SdsError sdsError) {
+            sdsError.print();
+            throw sdsError;
+        } catch (MalformedURLException mal) {
+            System.out.println("MalformedURLException");
```

Comment:
i guess we do this everywhere, but have we seen one of these?  Is there enough information is we are just saying "MalformedURLException" and that is it...  I would think we would want to include more information in debugging to know what the URL we sent was, probably the method call (maybe just need to printStackTrace()??)

Reply:
Probably just needs to be written up as a different issue, because that is outside the scope of this change

Reply:
I agree that this could be done better so I will make a WI for this